clang/lib/Basic/Targets/AMDGPU.cpp

```diff
@@ -83,6 +84,7 @@ const LangASMap AMDGPUTargetInfo::AMDGPUDefIsPrivMap = {
     llvm::AMDGPUAS::FLAT_ADDRESS, // ptr32_uptr
     llvm::AMDGPUAS::FLAT_ADDRESS, // ptr64
     llvm::AMDGPUAS::FLAT_ADDRESS, // hlsl_groupshared
+    llvm::AMDGPUAS::FLAT_ADDRESS, // hlsl_private
```

Comment:
I don't understand what hlsl_private means but using flat here is almost certainly wrong

Reply:
This address space will be used for:
 - static [const] global variables
Which matches the `Private` storage class for SPIR-V.

Would `LOCAL_ADDRESS` be a better match?

Reply:
Or maybe `PRIVATE_ADDRESS` would be best, since it's local per thread/wave

Reply:
If it acts like thread local storage, we don't have an implementation of that right now. So it would be none of these. PRIVATE_ADDRESS globals will just fail and nothing else has thread-local like behavior

Reply:
HLSL documentation for those is currently this: https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-variable-syntax

FXC/DXC implementation of a `static int my_global` would be equivalent to a c++ `static thread_local int my_global`. Initialized once, thread local.

If this is not available in AMDGPU yet, what value shall be picked?

Reply:
Probably should use PRIVATE_ADDRESS. It will at least fail in a way that makes it somewhat obvious what the issue is